src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/database/jpa/entity/induction/FutureWorkInterestsEntity.kt

```diff
@@ -132,7 +151,18 @@ class WorkInterestEntity(
   @Column
   @LastModifiedBy
   var updatedBy: String? = null,
-) : EntityKeyAware {
+) : KeyAwareChildEntity {
+
+  @PrePersist
+  @PreUpdate
+  @PreRemove
+  fun onChange() {
+    parent?.childEntityUpdated()
+  }
+
+  override fun associateWithParent(parent: ParentEntity) {
```

Comment:
This is called when adding this entity to the parent collection (via `InductionEntityListManager`). This is required for the bidirectional relationship to work (i.e. so that `parent?.childEntityUpdated()` can be called).